include/cinder/gl/GlslProg.h

```diff
@@ -45,11 +44,49 @@
 namespace cinder { namespace gl {
 	
 typedef std::shared_ptr<class GlslProg> GlslProgRef;
+	
+class UniformValueCache;
 
 class ShaderPreprocessor;
 
 class GlslProg : public std::enable_shared_from_this<GlslProg> {
   public:
+	
+	//! Resembles all the information Queryable of an attribute
+	struct Attribute {
```

Comment:
If these structs are going to be public, we should provide accessors instead of exposing public ivars with m prefix. Also, they look quite redendant, seems that it could be minimized by combining most of this into a base class (not sure what you'd call it..).